### PR TITLE
Improve settings

### LINKED ISSUE #133: Improve open file / directory dialogs in Settings
When a user changes their settings, the current behaviour of all open file / directory dialogs is to open in the CWD. If the user has already bound a setting to `some/very/very/very/very/long/path/executable` it would be nicer to open the file / directory dialog directory `some/very/very/very/very/long/path/`

### PR DESCRIPTION
**Pull request by snim2**
This PR improves the settings dialog. Previously, the file / directory open dialog always opened in the CWD. Following this PR, the dialog will check whether the relevant setting already has a value set, and if it does, will open in the same directory as the current setting.

If the relevant setting has was not set, the dialog opens in the `$HOME` directory.

This PR also adds the ruby-parser to the Settings dialog.

Fixes #133 
Fixes an issue discussed in #134

### CODE REVIEW

**Comment by snim2** [comment]
I think that last build failed due to the "randomised" tests, so I will restart it and see what happens.

File: lib/eco/rubyparser/rubyparser.py [hunk 1]
```diff
@@ -115,18 +116,22 @@ def _find_node(self, lineno, charno):
                 return node
 
     def inc_parse(self, line_indents = [], reparse=False):
+        settings = QSettings("softdev", "Eco")
+        ruby_parser =str(settings.value("env_ruby_parser", "").toString())
```
Comment (ltratt) [inline lib/eco/rubyparser/rubyparser.py]: Space missing after `=`.

**Comment by ltratt** [comment]
I wonder if we want to say "Ruby parser" rather than "JRuby parser"? I don't think there's anything JRuby specific here is there?

**Comment by snim2** [comment]
That means creating a new pane on the Settings dialog. Can do.

**Comment by ltratt** [comment]
Maybe we could have a Ruby pane with a JRuby sub-part of it?

**Comment by snim2** [comment]
Ready to squash?

New settings pane:

new_ruby_settings

**Comment by ltratt** [comment]
I don't get all of the icons on my machine (which isn't worth worrying about), but otherwise this looks good. Squash away!

**Comment by snim2** [comment]
All squashed!

**Comment by snim2** [comment]
Thanks

**Pull request merged**